build_ngtf.py

```diff
@@ -227,36 +260,75 @@ def main():
+                          "https://github.com/tensorflow/tensorflow.git",
+                          tf_version)
+            os.chdir(pwd_now)
+
+            # Next install the tensorflow python packge
             command_executor(
                 ["pip", "install", "-U", "tensorflow==" + tf_version])
             cxx_abi = get_tf_cxxabi()
+
+            # Copy the libtensorflow_framework.so to the artifacts so that
+            # we can run c++ tests from that location later
+            tf_fmwk_lib_name = 'libtensorflow_framework.so.1'
+            if (platform.system() == 'Darwin'):
+                tf_fmwk_lib_name = 'libtensorflow_framework.1.dylib'
+            import tensorflow as tf
+            tf_lib_dir = tf.sysconfig.get_lib()
+            tf_lib_file = os.path.join(tf_lib_dir, tf_fmwk_lib_name)
+            print("SYSCFG LIB: ", tf_lib_file)
+
+            dst_dir = os.path.join(artifacts_location, "tensorflow")
+            if not os.path.isdir(dst_dir):
+                os.mkdir(dst_dir)
+
+            dst = os.path.join(dst_dir, tf_fmwk_lib_name)
+            shutil.copyfile(tf_lib_file, dst)
+
+            # Now build the libtensorflow_cc.so - the C++ library
+            build_tensorflow_cc(tf_src_dir, artifacts_location, target_arch,
+                                verbosity)
+
         else:
-            if not arguments.skip_tensorflow_build:
-                print("Building TensorFlow")
-                # Download TensorFlow
-                download_repo("tensorflow",
-                              "https://github.com/tensorflow/tensorflow.git",
-                              tf_version)
-
-                # Build TensorFlow
-                build_tensorflow(venv_dir, "tensorflow", artifacts_location,
-                                 target_arch, verbosity)
-
-                # Install tensorflow
-                # Note that if gcc 4.8 is used for building TensorFlow this flag
-                # will be 0
-                cxx_abi = install_tensorflow(venv_dir, artifacts_location)
-            else:
-                cxx_abi = get_tf_cxxabi()
+            print("Building TensorFlow from source")
+            # Download TensorFlow
+            download_repo("tensorflow",
+                          "https://github.com/tensorflow/tensorflow.git",
+                          tf_version)
+            tf_src_dir = os.path.join(os.getcwd(), "tensorflow")
+            # Build TensorFlow
+            build_tensorflow(venv_dir, "tensorflow", artifacts_location,
+                             target_arch, verbosity)
+
+            # Now build the libtensorflow_cc.so - the C++ library
+            build_tensorflow_cc(tf_src_dir, artifacts_location, target_arch,
+                                verbosity)
+
+            # Install tensorflow to our own virtual env
+            # Note that if gcc 4.8 is used for building TensorFlow this flag
+            # will be 0
+            cxx_abi = install_tensorflow(venv_dir, artifacts_location)
```

Comment:
we can add an assert the comment that, `use_prebuilt_tensorflow implies cxx_abi==0`:
`assert (!arguments.use_prebuilt_tensorflow) or (arguments.use_prebuilt_tensorflow and cxx_abi==0), "Expected cxx_abi to be 0 when use_prebuilt_tensorflow is on"`